runtime/pkg/rduckdb/db.go

```diff
@@ -274,6 +274,15 @@ func NewDB(ctx context.Context, opts *DBOptions) (DB, error) {
 		}
 		return nil, err
 	}
+
+	// We want to prevent multiple rill process accessing same db files.
+	// All the files are accessed in read-only mode so it is possible for multiple rill process to access same db files.
+	// To prevent this we attach a dummy db file to the main in-memory db in write mode.
+	// This is required for local rill only but since there is no way to determine it in this package so we do it for all.
+	_, err = db.dbHandle.ExecContext(ctx, fmt.Sprintf("ATTACH %s AS __ymmud__", safeSQLString(filepath.Join(db.localPath, "main.db"))))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Instead of attaching, could it just pass `main.db` as the DSN to `openDBAndAttach`?